script.js

```diff
@@ -9,3 +10,247 @@ document.querySelectorAll('.head-link').forEach((n) => n.addEventListener('click
+    tecnologies: {
+      tech1: 'html',
+      tech2: 'Bootstrap',
+      tech3: 'Ruby on rails',
+    },
+    LinkToLive: 'https://tessyrich.github.io/Portfolio-Setup-and-mobile-first/',
+    ILinkToSource: 'https://github.com/Tessyrich/Portfolio-Setup-and-mobile-first',
+  },
+];
+
+const arrayOnclickBtn = document.querySelectorAll('.see1');
+
+const renderCards = (indx) => {
+  dataCards.map((card, cardIndx) => {
+    if (cardIndx === indx) {
+      document.querySelector('.pop-up').innerHTML = `
+      <div class="mult">
+      <div class="ham">
+        <h2 class="post post-small">${card.titlesmall}</h2>
+        <h2 class="post post-big">${card.title}</h2>
+        <span class="close-pop" id="close-pop" onclick='ClosePopUp()'>&times;</span>
+      </div>
+      <ul class="tag-main tagmain-details">
+        <li class="tag2 tagmain1"><a href="#">${card.tecnologies.tech1}</a></li>
+        <li class="tag3 tagmain2"><a href="#">${card.tecnologies.tech2}</a></li>
+        <li class="tag4 tagmain3"><a href="#">${card.tecnologies.tech3}</a></li>
+      </ul>
+    </div>
+    <div class="details-main">
+      <div class="details-img"><img src=${card.IMGURL} class="det-img"></div>
+      <div class="details-text">
+        <div class="text">
+          <p class="support sub-details">${card.description}</p>
+        </div>
+        <div class="but-details">
+          <section class="buttom buttom-first">
+            <a href=${card.LinkToLive} class="see">See live</a>
+            <img src="./images/icon-see live.png" alt="image icon see live">
+          </section>
+          <section class="buttom buttom-sec">
+            <a href=${card.LinkToSource} class="see">See source</a>
+            <img src="./images/Icon -GitHub.png" alt=" image icon Twitter">
+          </section>
+        </div>
+      </div>
+      </div>
+      `;
+    }
+    return dataCards;
+  });
+};
+for (let i = 0; i < arrayOnclickBtn.length; i += 1) {
+  arrayOnclickBtn[i].addEventListener('click', () => {
+    renderCards(i);
+    ClosePopUp();
+    popContainer.style.display = 'block';
+  });
+}
+
+// FORM VALIDATION
+const formVal = document.querySelector('.form');
+const inp1 = document.querySelectorAll('.inp1');
+const inp2 = document.querySelectorAll('.inp2');
+const inp3 = document.querySelectorAll('.inp3');
+const emailEl = document.querySelector('#email');
+
+const oluchiValidate = (inputbox) => {
+  if (inputbox !== '') {
+    return inputbox;
+  }
+  return 'please insert your name';
+};
+
+formVal.addEventListener('submit', (e) => {
+  e.preventDefault();
+  oluchiValidate(inp1.value);
```

Comment:
- [ ] As reqquired by the previous reviewer, I kindly suggest that you use a better name for your function here name `oluchiValidate`